tiledb/sm/c_api/tiledb.cc

```diff
@@ -3260,12 +3260,9 @@ int32_t tiledb_query_get_array(
     return TILEDB_OOM;
   }
 
-  // Allocate an array object, copied from the query's array.
-  (*array)->array_ =
-      new (std::nothrow) tiledb::sm::Array(*query->query_->array());
+  // Allocate an array object, taken from the query's array.
+  (*array)->array_ = query->query_->array_shared();
   if ((*array)->array_ == nullptr) {
```

Comment:
With the elimination of the call to `new` there's no longer a possibility for an error here. This whole `if` statement and its body is redundant.